packages/babel-parser/test/fixtures/typescript/cast/unparenthesized-assert-and-assign/output.json

```diff
@@ -3,8 +3,7 @@
   "start":0,"end":46,"loc":{"start":{"line":1,"column":0,"index":0},"end":{"line":2,"column":29,"index":46}},
   "errors": [
     "SyntaxError: Invalid left-hand side in assignment expression. (1:0)",
-    "SyntaxError: Invalid left-hand side in assignment expression. (2:6)",
-    "SyntaxError: Invalid left-hand side in object destructuring pattern. (2:6)"
+    "SyntaxError: Invalid left-hand side in assignment expression. (2:6)"
```

Comment:
These two errors were at the same location.

It would be better to only keep the "object destructuring pattern" rather than "assignment expression", but it's quite complex because it requires delaying errors until we know what the outer object is (destructuring or object). I might improve it in a follow-up PR.